### PR TITLE
feat(AIP-2): add AEP around numbering.

### PR DESCRIPTION
**Pull request by toumorokoshi**
Adding an AEP about numbering enables
a clear organization for future contributions.

Modification of existing AEPs to the new numbering scheme will
be added in future commits.

### CODE REVIEW

File: aep/general/0151/lro.oas.yaml
Comment (rofrankel) [inline aep/general/0151/lro.oas.yaml:25]: Does this PR need to change this file?

File: aep/general/0002/aep.md
Comment (rofrankel) [inline aep/general/0002/aep.md:19]: I thought we had tentatively concluded that this was duplicative of **must** vs. **should**, and so maybe we weren't going to do it?

File: aep/general/0151/lro.oas.yaml
Comment (toumorokoshi) [reply aep/general/0151/lro.oas.yaml:25]: nope sorry, don't know what happened there.

File: aep/general/0002/aep.md
Comment (toumorokoshi) [reply aep/general/0002/aep.md:19]: Hmm... wires crossed? I definitely don't see the distinction as redundant: we should point at which AEPs one should strictly adhere to.

I don't think this would work otherwise: e.g. if we decided on singletons being a design pattern, are you okay with every single clause in that AEP being a **should**?

Comment (rofrankel) [reply aep/general/0002/aep.md:19]: Discussed in weekly meeting, but for posterity:  we discussed this live and decided to go with a different model (something more like AEP states, plus drafts of new AEPs being able to replace existing AEPs).

Marking thread resolved since it's obsolete.

**Comment by toumorokoshi** [comment]
@rofrankel ready for another review!

File: aep/general/0002/aep.md
Comment (rofrankel) [inline aep/general/0002/aep.md:17]: Nit:  the number has to be assigned before merging, right?

File: aep/general/0002/aep.md
Comment (rofrankel) [inline aep/general/0002/aep.md:23]: I think this addresses the numbering question, but is light on details about what it means to extend the AEPs and how to do it.  Maybe this AEP doesn't need to cover those details, but it would be helpful for it to at least link to something that does.

File: aep/general/0002/aep.md
Comment (rofrankel) [inline aep/general/0002/aep.md:28]:
```suggestion
To help with discovery of an AEP, each one is assigned a category: a string that
```

File: aep/general/0002/aep.md
Comment (rofrankel) [inline aep/general/0002/aep.md:29]: "proposal" as in the "P" in "AEP"?  Should we just say AEP here?

File: aep/general/0002/aep.md
Comment (rofrankel) [inline aep/general/0002/aep.md:34]: I don't think this is actually ungrammatical, but it reads a little awkwardly - it uses "by which" but still manages to end in a preposition!  Maybe something like this?

```suggestion
Each AEP is given a slug: a string by which it can be easily referenced. The
```

File: aep/general/0002/aep.md
Comment (rofrankel) [inline aep/general/0002/aep.md:36]: This section could use examples, both of a slug changing to refer to a new revision of an AEP, but also of what a slug even is.

File: aep/general/0002/aep.md
Comment (rofrankel) [inline aep/general/0002/aep.md:49]: Should we use categories (many:one) or tags (many:many)?

File: aep/general/0002/aep.md
Comment (toumorokoshi) [reply aep/general/0002/aep.md:29]: I thought it might be better to have a separate term for a single AEP, and thought "proposal" might work. In case we decide to change names again.

But I could just use AEP everywhere, if you feel strongly about it.

File: aep/general/0002/aep.md
Comment (toumorokoshi) [reply aep/general/0002/aep.md:17]: how about "as it's merged"?

File: aep/general/0002/aep.md
Comment (toumorokoshi) [reply aep/general/0002/aep.md:49]: I was thinking categories to start: it gives us the ability for a clear ordering in the left-hand-navigation.

If we only had tags then in the nav you'd have the same AEP show up multiple times - I think that might be make the tree hierarchy more difficult to navigate?

File: aep/general/0002/aep.md
Comment (toumorokoshi) [reply aep/general/0002/aep.md:23]: There's already a lot of content, so I'd rather do that incrementally if that's ok. Tracking ticket: https://github.com/aep-dev/aep.dev/issues/69.

File: aep/general/0002/aep.md
Comment (toumorokoshi) [reply aep/general/0002/aep.md:36]: added.

File: aep/general/0002/aep.md
Comment (toumorokoshi) [reply aep/general/0002/aep.md:34]: applied!

File: aep/general/0002/aep.md
Comment (toumorokoshi) [reply aep/general/0002/aep.md:29]: I just updated everything to say "AEP" explicitly.

File: aep/general/0002/aep.md
Comment (toumorokoshi) [reply aep/general/0002/aep.md:17]: updated to use "as it's merged" language, PTAL.

File: aep/general/0002/aep.md
Comment (rofrankel) [inline aep/general/0002/aep.md:56]: Are slugs also used for short links, e.g. aep.dev/create?

**Review by rofrankel — APPROVED** [review]
LGTM, but added one optional comment to look at.

File: aep/general/0002/aep.md
Comment (toumorokoshi) [reply aep/general/0002/aep.md:56]: I was thinking of adding that once I do work in the site-generator to support it, but I don't mind adding that now. WDYT?

If we did, I think `aep.dev/{slug}` makes sense. Although we may need to make sure slug names can't be used (e.g. `category`).

I'd kind of wonder if we want to namespace this, with something like `aep.dev/p/create` or something like that. A little less aesthetically pleasing, but works around a giant headache?

Comment (rofrankel) [reply aep/general/0002/aep.md:56]: Might as well write it now IMO, it's not like the site generator is in tightly sync with this repo anyway.  For now we can implement it as a matter of policy - the shortlink (already supported by the site generator) must match the slug.

I think the `/p/` gets too much in the way of typing out shortlinks like `aip.dev/create`.  Better to namespace categories, if we're worried about collisions, IMO - they'll probably be much less frequently referenced by manually typed links.

Comment (toumorokoshi) [reply aep/general/0002/aep.md:56]: ack, will add it before I merge.

I was more concerned about having the set of possible slugs be everything, except for a set of constants (e.g. "categories" so we don't conflict with them). 

Just playing around with this a bit, I think we're looking at:

```
aep.dev/ # landing page
aep.dev/create # slug to an AEP
aep.dev/2 # the actual AEP.
aep.dev/categories/ # category page
aep.dev/categories/resources # page listing AEPs for a category
```

Comment (rofrankel) [reply aep/general/0002/aep.md:56]: SGTM, approving to unblock the PR whenever you make that change.

**Review by rofrankel — APPROVED** [review]

File: aep/general/0002/aep.md
Comment (toumorokoshi) [reply aep/general/0002/aep.md]: added!

**Pull request merged**
